src/Compilers/VisualBasic/Portable/Binding/Binder_Symbols.vb

```diff
@@ -516,10 +511,15 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 Next
 
                 If forwardedType IsNot Nothing Then
-                    If forwardedType.IsErrorType Then
-                        Dim diagInfo As DiagnosticInfo = DirectCast(forwardedType, ErrorTypeSymbol).ErrorInfo
-                        If diagInfo.Code = ERRID.ERR_TypeFwdCycle2 Then
-                            encounteredCycle = True
+                    If Not reportedAnError And forwardedType.IsErrorType Then
```

Comment:
From what I can tell in the rest of the binder code, both are passed around for various checks. Do we want to make the exception here (changing it to use a Nothing bag if reportedAnError was true, doing the null check before reporting?).